cmd/common/utils/parser-tableformatter.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+
+	"github.com/kinvolk/inspektor-gadget/pkg/columns"
+	"github.com/kinvolk/inspektor-gadget/pkg/columns/formatter/textcolumns"
+)
+
+const (
+	KubernetesTag       string = "kubernetes"
+	ContainerRuntimeTag string = "runtime"
+)
+
+var SupportedTags = map[string]struct{}{
+	KubernetesTag:       {},
+	ContainerRuntimeTag: {},
+}
+
+type Option func(*GadgetOptions) error
+
+func WithMetadata(metadataTag string) Option {
+	return func(opts *GadgetOptions) error {
+		if _, ok := SupportedTags[metadataTag]; !ok {
+			return fmt.Errorf("invalid metadata tag: %q", metadataTag)
+		}
+		opts.metadataTag = metadataTag
+		return nil
+	}
+}
+
+type GadgetOptions struct {
+	metadataTag string
+}
+
+// GadgetParser is a parser helps printing the gadget output in columns using
+// the columns and tableformatter packages.
+type GadgetParser[T any] struct {
+	columns   *columns.Columns[T]
+	formatter *textcolumns.TextColumnsFormatter[T]
+}
+
+func NewGadgetParser[T any](outputConfig *OutputConfig, options ...Option) (*GadgetParser[T], error) {
+	var opts GadgetOptions
+
+	for _, o := range options {
+		err := o(&opts)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	cols, err := columns.NewColumns[T]()
+	if err != nil {
+		return nil, fmt.Errorf("creating columns: %w", err)
+	}
+
+	// If no tag is provided, we use only the columns with no specific tag. In
+	// other words, the gadget-specific columns. Otherwise, we also include the
+	// columns with the requested tag.
+	filteredColumns := cols.GetColumnMap(columns.WithNoTags())
+	if opts.metadataTag != "" {
+		filteredColumns = cols.GetColumnMap(columns.Or(columns.WithTag(opts.metadataTag), columns.WithNoTags()))
+	}
```

Comment:
Perhaps this is a bit more efficient and easier to understand:
```suggestion
	var filter columns.ColumnFilter
	if opts.metadataTag == "" {
		filter = columns.WithNoTags()
	} else {
		filter = columns.Or(columns.WithTag(opts.metadataTag), columns.WithNoTags())
	}

	filteredColumns := cols.GetColumnMap(filter)
```

Also `filteredColumns` could be removed and could be called directly when creating the parser?